packages/compiler-sfc/src/compileScript.ts

```diff
@@ -832,8 +840,19 @@ export function compileScript(
   }
 
   // 10. generate return statement
-  let returned
-  if (
+  let returned = ''
+  if (ctx.renderFunction) {
+    if (sfc.template) {
```

Comment:
It seems that defineRender has a higher priority. I don’t know if this needs to be consistent with the current version priority (the template and the rendering function returned by setup exist at the same time), but I think the current version should also throw a warning.

Reply:
There will be a meaningful return statement in only two conditions.
- `sfc.template`: warned in line#846. Should we warn or throw an error directly?
- `!options.inlineTemplate`: `defineRender` indeed has higher priority than it. We cannot have both (return setup bindings and render function) at the same time, which could be bad for devtools DX. But it's a trade-off.